### DISCUSSION TITLE
Getting started / deploy process

### DISCUSSION THREAD

**Discussion by JackGilmore** [timwis/jkan#200] [Ideas]
Continued from #198 

I had wondered whether it's worth changing the setup process from forking to setting up the upstream repository as a template repository instead? Given that we'll all have different datasets and orgs we're hosting, I didn't really see a scenario in which forking makes sense for contributing upstream from our own JKAN instances?

Whilst I've not looked in detail, if I understand template repos correctly, I think it's still possible to pull downstream changes from the original template into your instance if you wanted to pull in new changes and bug fixes? This was a short article I found on the topic https://medium.com/geekculture/how-to-use-git-to-downstream-changes-from-a-template-9f0de9347cc2

**Comment by timwis** [comment]
Sounds like an interesting idea! I haven't used GitHub templates, so I'm not entirely familiar with how they work.

> I didn't really see a scenario in which forking makes sense for contributing upstream from our own JKAN instances?
I agree - it's more-so there for pulling upstream changes downward (e.g. a new feature, or patches to dependencies).

The only advantages I can think of regarding forks are that:

1. We have a crude way of seeing who's using JKAN (although admittedly, most forks are just someone poking around with it, and several of the biggest users don't technically use a fork)
2. There's a nifty little feature in `/admin` that lets you check for updates by linking to a pull request between upstream and your repo.

I don't think either of those points are worth hanging on to forks if templates offer an advantage. But we'd probably want to make sure you can still at least pull in updates from upstream. From reading over the templates documentation, I'm not sure whether that's possible. Do you know?

**Reply by JackGilmore** [reply]
I'm not sure if that's a supported feature but from the brief look I had over the article I posted above, you can use git commands to grab updates from the upstream. I'll have a look into this further and see how easy it is...
